music21/musicxml/m21ToXml.py

```diff
@@ -2707,15 +2702,19 @@ def fixupNotationFlat(self):
         '''
         Runs makeNotation on a flatStream, such as one lacking measures.
         '''
+        # Do this before makeNotation so that measures are filled correctly
+        self.stream = self.stream.splitAtDurations(recurse=True)[0]
+
         part = self.stream
         part.makeMutable()  # must mutate
         # try to add measures if none defined
         # returns a new stream w/ new Measures but the same objects
         part.makeNotation(meterStream=self.meterStream,
                           refStreamOrTimeRange=self.refStreamOrTimeRange,
                           inPlace=True)
-        # environLocal.printDebug(['fixupNotationFlat: post makeNotation, length',
-        #                    len(measureStream)])
+
+        # Do this again, since makeNotation() might create complex rests
+        self.stream = self.stream.splitAtDurations(recurse=True)[0]
```

Comment:
It's fairly rare to go through `fixupNotationFlat()`, since even "flat" scores go through a "general object conversion". You'd need a not-well-formed score or makeNotation=False to end up here. So that's why I thought this was not terrible to do this twice (better than erroring out with complex duration failures).

Reply:
Unfortunately, many of the musicxml unit tests skip the general object conversion. When I get some time I might do a tiny PR to add a `makeNotation=True` default to the `getET` test utility.

Reply:
Actually, it looks like I factored this a little differently in #1242, so maybe that's worth reviewing first.